pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java

```diff
@@ -615,6 +621,12 @@ private static void runProducer(int producerId,
                     }
                     if (arguments.delay > 0) {
                         messageBuilder.deliverAfter(arguments.delay, TimeUnit.SECONDS);
+                    } else {
+                        if (arguments.delayRange != null) {
+                            final long deliverAfter = ThreadLocalRandom.current()
+                                    .nextLong(arguments.delayRange.lowerEndpoint(), arguments.delayRange.upperEndpoint());
```

Comment:
Excluding upperEndpoint?

Reply:
Note: `nextLong()` including lowerEndpoint and excluding upperEndpoint

Reply:
fixed